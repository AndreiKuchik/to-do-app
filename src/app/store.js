import { configureStore } from '@reduxjs/toolkit'

import toDoListReducer from '../features/toDoListSlice'

export default configureStore({
  reducer: {
    toDoList: toDoListReducer
  },
})
