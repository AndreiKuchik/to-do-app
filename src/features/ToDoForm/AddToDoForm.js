import React, { useState } from 'react';
import { unwrapResult } from '@reduxjs/toolkit';
import { useAppDispatch } from '../../core/hooks';
import styles from './AddToDoFrom.module.css';
import { addNewToDo } from '../toDosSlice';
import Card from '../../components/Card/Card';

const AddPostForm = () => {
  const [toDo, setToDo] = useState('');
  const dispatch = useAppDispatch();

  const onToDoChanged = (e) => setToDo(e.target.value);

  const canSave = !(typeof toDo === 'string' && toDo.trim().length === 0);

  const onSavePostClicked = async () => {
    if (canSave) {
      try {
        const resultAction = await dispatch(addNewToDo(toDo));
        unwrapResult(resultAction);
        setToDo('');
      } catch (err) {
        console.log('Failed to save the ToDo: ', err);
      }
    }
  };

  return (
    <Card className={styles.toDoForm}>
      <form>
        <div className={styles.newToDoControls}>
          <div className={styles.newToDoControl}>
            <input
              className = 'todo-input'
              type="text"
              value={toDo}
              onChange={onToDoChanged}
              placeholder="Please type your ToDO"
            />
          </div>
          <div className={styles.newToDoActions}>
            <button
            className='submit-todo-input'
              type="button"
              onClick={onSavePostClicked}
              disabled={!canSave}
            >
              Save ToDo
            </button>
          </div>
        </div>
      </form>
    </Card>
  );
};

export default AddPostForm;
