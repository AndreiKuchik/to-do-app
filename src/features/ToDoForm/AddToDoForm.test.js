import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import AddToDoForm from './AddToDoForm';
import store from '../../core/store';

describe('Testing AddToDoForm component', () => {
  it('Matches DOM Snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <AddToDoForm />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
