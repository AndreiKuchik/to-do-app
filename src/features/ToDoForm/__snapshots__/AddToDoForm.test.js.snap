// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddToDoForm component Matches DOM Snapshot 1`] = `
<div>
  <div
    class="card toDoForm"
  >
    <form>
      <div
        class="newToDoControls"
      >
        <div
          class="newToDoControl"
        >
          <input
            class="todo-input"
            placeholder="Please type your ToDO"
            type="text"
            value=""
          />
        </div>
        <div
          class="newToDoActions"
        >
          <button
            class="submit-todo-input"
            disabled=""
            type="button"
          >
            Save ToDo
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
